### PR TITLE
BUG: GH15426 timezone lost in groupby-agg with cython functions

### LINKED ISSUE #15426: BUG: timezone lost in groupby-agg with cython functions
xref #10668 (for more examples)

Hello!

I'm running into some odd behavior trying to group rows of a pandas dataframe by ID and then selecting out max/min datetimes (w/ timezones). This is with python 2.7, pandas 0.18.1 and numpy 1.11.1 (I saw in earlier posts a similar problem was apparently fixed w/ pandas 0.15).

Specifically, if I try:
`print orders.groupby('OrderID')['start_time'].agg(np.min).iloc[:5]`

I get:
```
OrderID
O161101XVS100000044   2016-11-01 12:03:12.920000-04:00
O161101XVS100000047   2016-11-01 12:03:36.693000-04:00
O161101XVS100000098   2016-11-01 12:09:08.330000-04:00
O161101XVS100000122   2016-11-01 12:09:59.950000-04:00
O161101XVS100000152   2016-11-01 12:11:29.790000-04:00
Name: start_time, dtype: datetime64[ns, US/Eastern]
```

Where the raw data had times closer to 8 am (US/Eastern). In other words, it reverted back to UTC times, even though it says it's eastern times, and has UTC-4 offset.

But if I instead try:
`print orders.groupby('OrderID')['start_time'].agg(lambda x: np.min(x)).iloc[:5]`

I now get:
```
OrderID
O161101XVS100000044   2016-11-01 08:03:12.920000-04:00
O161101XVS100000047   2016-11-01 08:03:36.693000-04:00
O161101XVS100000098   2016-11-01 08:09:08.330000-04:00
O161101XVS100000122   2016-11-01 08:09:59.950000-04:00
O161101XVS100000152   2016-11-01 08:11:29.790000-04:00
Name: start_time, dtype: datetime64[ns, US/Eastern]
```

Which is the behavior I intended. This second method is vastly slower, and I would have assumed the two approaches would yield identical results ...

### PR DESCRIPTION
**Pull request by stephenrauch**
- [x] closes #15426
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_aggregate.py [hunk 1]
```diff
@@ -738,3 +738,14 @@ def test_agg_over_numpy_arrays(self):
                                 columns=expected_column)
 
         assert_frame_equal(result, expected)
+
+    def test_agg_time_zone_round_trip(self):
+        ts = pd.Timestamp("2016-01-01 12:00:00", tz='US/Pacific')
```
Comment (jreback) [inline pandas/tests/groupby/test_aggregate.py]: add the issue number as a comment

File: pandas/tests/groupby/test_aggregate.py [hunk 2]
```diff
@@ -738,3 +738,14 @@ def test_agg_over_numpy_arrays(self):
...
+        df = pd.DataFrame({'a': 1, 'b': [ts + timedelta(minutes=nn)
+                                         for nn in range(10)]})
+
+        result1 = df.groupby('a')['b'].agg(np.min).iloc[0]
```
Comment (jreback) [inline pandas/tests/groupby/test_aggregate.py]: compare with ``.min()`` as well

File: pandas/types/cast.py [hunk 3]
```diff
@@ -133,7 +133,8 @@ def trans(x):  # noqa
                 if dtype.tz:
                     # convert to datetime and change timezone
                     from pandas import to_datetime
-                    result = to_datetime(result).tz_localize(dtype.tz)
+                    result = to_datetime(result).tz_localize('utc')
```
Comment (jreback) [inline pandas/types/cast.py]: can add addtl tests in ``pandas/tests/types/test_cast.py`` fit them in as appropriate. take out this fix temporarily to validate your test cases, then add back to make sure they work.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
add examples from https://github.com/pandas-dev/pandas/issues/10668 as well. (and you can list list issue in the whatsnew too)

**Comment by jreback** [comment]
can you update

**Comment by stephenrauch** [comment]
I will take care of this this weekend...

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**